### DISCUSSION TITLE
How can I declare a singular value in the body request that is compatible with requests?

### DISCUSSION THREAD

**Discussion by jalvespinto** [fastapi/fastapi#5025] [Questions] [answered]
I have a route that takes a single body parameter of type list[str]:

@app.post("/")
def endpoint(param1: list[str]: Body(default=None)):
    ...

This way the route expect a body like ["string"]. But now I don't know how to pass the body parameter on a request, since requests don't take lists as an valid body. A simple work around would be to create a schema, but that would make me have to write a little bit more code on the body of the function. Not a big problem, but I was wondering if I am doing something wrong.

**Comment by odiseo0** [comment, marked as answer]
I don't fully understand your question, can you elaborate?

If you want to send a `list` of `str` as a Body using `requests` you can do it with the `json` module
```py
import requests
import json

requests.post("url", data=json.dumps(["strings", "strings"]))
requests.post("url", json=json.dumps(["strings", "strings"]))
```

**Reply by jalvespinto** [reply]
tks. Actually you already answered my question. I was trying something like data={"param1": ["string"]} and was getting 422 error.
